src/server/modules/code-action.ts

```diff
@@ -222,23 +221,25 @@ export class CodeActionModule implements LanguageServerModule {
 
 			const action = await this.#getAutoFixAllAction(document);
 
-			return action ? [action] : [];
+			if (action) {
+				fixAllActions.push(action);
+			}
 		}
 
 		if (only?.has(LSP.CodeActionKind.Source)) {
 			this.#logger?.debug('Creating "source" code action');
 
-			return [this.#getAutoFixAllCommandAction(document)];
+			fixAllActions.push(this.#getAutoFixAllCommandAction(document));
 		}
 
 		if (only && !only.has(LSP.CodeActionKind.QuickFix)) {
 			this.#logger?.debug('No quick fix actions requested, skipping action creation');
```

Comment:
I think this log will be output even if we have fixAllActions. Could you please check the following conditions before outputting the log?

```js
!only.has(LSP.CodeActionKind.SourceFixAll) &&
!only.has(StylelintCodeActionKind.StylelintSourceFixAll) &&
!only.has(LSP.CodeActionKind.Source)
```

Reply:
Indeed it will :thinking:

If I add these conditions, we'll only skip creating quick fixes if none of our supported CodeActions are provided.

A client could request:
- FixAll actions but no quick fixes,
- quickfix actions but no FixAll?

Could we add several specific debug outputs? I'll push a commit to make it clearer but I'll refactor the way you want if you don't agree :+1: